src/experimental/tools/mediaCapabilitiesProber/probers/__tests__/decodingInfo.test.ts

```diff
@@ -106,69 +103,73 @@ describe("MediaCapabilitiesProber probers - decodingInfo", () => {
+      expect(decodingInfoStub).not.toHaveBeenCalled();
+      resetDecodingInfos();
+    }
+    expect(thrownException).toEqual(true);
   });
 
-  it("should throw if no type in config", () => {
+  it("should throw if no type in config", async () => {
     const decodingInfoStub = stubDecodingInfo(true);
     const configuration = {
       audio: {
         contentType: "audio/wma",
       },
     };
-    expect.assertions(2);
-    return probeDecodingInfos(configuration)
-      .then(() => {
-        resetDecodingInfos();
-      })
-      .catch(({ message }: { message: string }) => {
-        expect(message).toEqual(
-          "MediaCapabilitiesProber >>> API_CALL: " +
-            "Not enough arguments for calling mediaCapabilites.",
-        );
-        expect(decodingInfoStub).not.toHaveBeenCalled();
-        resetDecodingInfos();
-      });
+    let thrownException = false;
+    try {
+      await probeDecodingInfos(configuration);
+      resetDecodingInfos();
+    } catch (error) {
+      const message = error instanceof Error ? error.message : "";
+      thrownException = true;
+      expect(message).toEqual("Not enough arguments for calling mediaCapabilites.");
+      expect(decodingInfoStub).not.toHaveBeenCalled();
+      resetDecodingInfos();
+    }
+    expect(thrownException).toEqual(true);
   });
 
-  it("should throw if empty config", () => {
+  it("should throw if empty config", async () => {
     const decodingInfoStub = stubDecodingInfo(true);
     const configuration = {};
 
-    expect.assertions(2);
-    return probeDecodingInfos(configuration)
-      .then(() => {
-        resetDecodingInfos();
-      })
-      .catch(({ message }: { message: string }) => {
-        expect(message).toEqual(
-          "MediaCapabilitiesProber >>> API_CALL: " +
-            "Not enough arguments for calling mediaCapabilites.",
-        );
-        expect(decodingInfoStub).not.toHaveBeenCalled();
-        resetDecodingInfos();
-      });
+    let thrownException = false;
+    try {
+      await probeDecodingInfos(configuration);
+      resetDecodingInfos();
+    } catch (error) {
+      const message = error instanceof Error ? error.message : "";
+      thrownException = true;
+      expect(message).toEqual("Not enough arguments for calling mediaCapabilites.");
+      expect(decodingInfoStub).not.toHaveBeenCalled();
+      resetDecodingInfos();
+    }
+    expect(thrownException).toEqual(true);
   });
 
   it("should throw if API mediaCapabilities not available", () => {
     // @ts-expect-error: `navigator.mediaCapabilities` is read-only normally, for
     // now, we're going through JSDom through so that's OK.
     delete navigator.mediaCapabilities;
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    expect(probeDecodingInfos({})).rejects.toThrowError(
-      "MediaCapabilitiesProber >>> API_CALL: MediaCapabilities API not available",
+    return probeDecodingInfos({}).then(
```

Comment:
why is there a `return` ?
Also there is jest modifier `.rejects` that help with this kind of test:
```js
await expect(probeDecodingInfos({})).rejects.toThrowError("navigator.mediaCapabilites.decodingInfo is not available");
```

Reply:
When tests return a promise in most testing frameworks, the test will resove when the promise rejects and reject when the promise resolve.

I know about `rejects`, and that was what was there before, but for some reason I found returning the promise more readable.